File: apps/harness/lib/faceStyle.ts
/**
 * Cluster face style — layout only; protocol JSON fields stay frozen.
 *
 * ap1 (default): locked flat elevation — straight TEMP / FUEL.
 * ap2: interpretive arched side-gauges from refs/oem/ap2/ (not pixel-perfect).
 */

export const FACE_STYLES = ["ap1", "ap2"] as const;
export type FaceStyle = (typeof FACE_STYLES)[number];
export const DEFAULT_FACE_STYLE: FaceStyle = "ap1";

export const FACE_STYLE_LABELS: Record<FaceStyle, string> = {
  ap1: "AP1",
  ap2: "AP2",
};

export const FACE_STYLE_HINTS: Record<FaceStyle, string> = {
  ap1: "Straight TEMP / FUEL — locked flat elevation",
  ap2: "Arched side gauges — interpretive, not a measured plate",
};

export function parseFaceStyle(value: string | null | undefined): FaceStyle {
  const token = (value ?? "").trim().toLowerCase();
  if (token === "ap2") return "ap2";
  return "ap1";
}

File: src/face_style.py
"""Cluster face style — layout only; protocol fields stay frozen.

AP1 (default): locked flat elevation — straight TEMP / FUEL, OEM telltale strip.
AP2: interpretive arched side-gauges (TEMP over FUEL on the right) plus a
clock row. Geometry is *not* a pixel-perfect AP2 plate — see refs/oem/ap2/.

Future chassis can add more members without renaming JSON fields.
"""
from __future__ import annotations

from enum import Enum


class FaceStyle(str, Enum):
    AP1 = "ap1"
    AP2 = "ap2"


DEFAULT_FACE_STYLE = FaceStyle.AP1

FACE_STYLE_LABELS: dict[FaceStyle, str] = {
    FaceStyle.AP1: "AP1 — straight TEMP / FUEL",
    FaceStyle.AP2: "AP2 — arched side gauges",
}


class FaceStyleError(ValueError):
    """Unknown or empty face style."""


def parse_face_style(value: str | FaceStyle | None) -> FaceStyle:
    """Parse a style token. Blank / None → AP1 default."""
    if value is None or value is FaceStyle.AP1:
        return FaceStyle.AP1
    if isinstance(value, FaceStyle):
        return value
    token = str(value).strip().lower()
    if not token:
        return FaceStyle.AP1
    try:
        return FaceStyle(token)
    except ValueError as exc:
        known = ", ".join(s.value for s in FaceStyle)
        raise FaceStyleError(f"unknown face style {value!r}; expected {known}") from exc
